### PR TITLE
Benchmarks: Revise Test - Revise  benchmark test util to support pytorch multi-GPU test

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Revise benchmark test util to support multi-GPU test

**Major Revision**
-  simulate Pytorch DDP environment in benchmark test util
-  Add multi-processing logic for multi-GPU benchmark execution and collect results

**Minor Revision**
- Add init.py in 'tests' to identify the 'tests' module

### CODE REVIEW

File: tests/benchmarks/utils.py [hunk 1]
```diff
@@ -4,15 +4,11 @@
...
-    os.environ['WORLD_SIZE'] = '1'
-    os.environ['RANK'] = '0'
-    os.environ['LOCAL_RANK'] = '0'
-    os.environ['MASTER_ADDR'] = 'localhost'
-    os.environ['MASTER_PORT'] = '12345'
+import socket
+from contextlib import closing
+import multiprocessing as multiprocessing
+from multiprocessing import Process
+from superbench.benchmarks import BenchmarkRegistry
```
Comment (guoshzhao) [inline tests/benchmarks/utils.py:11]: Add blank line to separate native package and custom package. The common rule is:

import native package
blank line
import thirdparty pacakge
blank line
import custom package

File: tests/benchmarks/utils.py [hunk 2]
```diff
@@ -22,3 +18,49 @@ def clean_simulated_ddp_distributed_env():
     os.environ.pop('LOCAL_RANK')
     os.environ.pop('MASTER_ADDR')
     os.environ.pop('MASTER_PORT')
+
+
+def get_free_port():
+    """Get free port."""
```
Comment (guoshzhao) [inline tests/benchmarks/utils.py:24]: Add return description in docstring.

File: tests/benchmarks/utils.py [hunk 3]
```diff
@@ -22,3 +18,49 @@ def clean_simulated_ddp_distributed_env():
...
+    setup_simulated_ddp_distributed_env(world_size, local_rank, port)
+    benchmark = BenchmarkRegistry.launch_benchmark(context)
+    # parser object must be removed becaues it can not be serialized.
+    benchmark._parser = None
+    queue.put(benchmark)
+    clean_simulated_ddp_distributed_env()
+
+
+def simulated_ddp_distributed_benchmark(context, world_size):
+    """Function to run the benchmark on #world_size number of processes."""
```
Comment (guoshzhao) [inline tests/benchmarks/utils.py:51]: Add return description.

**Review by guoshzhao — APPROVED** [review]

**Review by cp5555 — APPROVED** [review]

File: tests/benchmarks/utils.py [hunk 4]
```diff
@@ -22,3 +18,49 @@ def clean_simulated_ddp_distributed_env():
...
+
+def get_free_port():
+    """Get free port."""
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+        s.bind(('', 0))
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        return s.getsockname()[1]
+
+
+def setup_simulated_ddp_distributed_env(world_size, local_rank, port):
```
Comment (abuccts) [inline tests/benchmarks/utils.py:31]: these functions are all test fixtures, could we use [`pytest.fixture`](https://docs.pytest.org/en/stable/unittest.html#mixing-pytest-fixtures-into-unittest-testcase-subclasses-using-marks) for such purpose?

File: tests/benchmarks/utils.py (hunk 1 above)
Comment (yukirora) [reply tests/benchmarks/utils.py:12]: ok, thank you very much

File: tests/benchmarks/utils.py (hunk 2 above)
Comment (yukirora) [reply tests/benchmarks/utils.py]: ok, thank you.

File: tests/benchmarks/utils.py (hunk 3 above)
Comment (yukirora) [reply tests/benchmarks/utils.py]: ok, thank you.

**Pull request merged**